projects/kalidea/kaligraphi/src/lib/atoms/kal-loader/kal-loading.directive.ts

```diff
@@ -0,0 +1,80 @@
+import { Directive, ElementRef, Input } from '@angular/core';
+import { Overlay, OverlayConfig, OverlayRef, PositionStrategy } from '@angular/cdk/overlay';
+import { ComponentPortal } from '@angular/cdk/portal';
+
+import { Coerce } from '../../utils/decorators/coerce';
+import { KalLoaderComponent } from './kal-loader.component';
+
+@Directive({
+  selector: '[kalLoading]'
+})
+export class KalLoadingDirective {
+
+  private overlayRef: OverlayRef;
+
+  private _loading;
+
+  constructor(private elementRef: ElementRef,
+              private overlay: Overlay) {
+  }
+
+  @Input('kalLoading')
+  @Coerce('boolean')
+  set loading(loading: boolean) {
+    if (loading === true) {
+      this.attachLoader();
+    } else if (this._loading === true) {
+      this.getOverlayRef().detach();
+    }
+    this._loading = loading;
+  }
+
+  private attachLoader() {
+    const portal = new ComponentPortal(KalLoaderComponent);
```

Comment:
this.getOverlayRef().attach(new ComponentPortal(KalLoaderComponent));

Reply:
this proposal reduce clarity